File: FlashCards.Data/Models/Lesson.cs
using System;
using System.Collections;
using System.Collections.Generic;

namespace FlashCards.Data.Models
{
    public class Lesson
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public string Category { get; set; }

        public DateTime DateCreated { get; set; }

        public DateTime DateModified { get; set; }


        public int CourseId { get; set; }

        public Course Course { get; set; }

        public ICollection<Flashcard> Flashcards { get; set; }
    }
}

File: FlashCards-SPA/src/app/core/_models/_dtos/fromServer/lesson.ts
import {FlashcardShort} from './flashcardShort';

export class Lesson {
  id: number;
  subLessonId: number;
  subCourseId: number;
  name: string;
  description: string;
  category: string;
  dateCreated: Date;
  dateModified: Date;
  isSubscribed: boolean;
  overallProgress: number;
  flashcards: FlashcardShort[];

  constructor(id: number, subLessonId: number, subCourseId: number, name: string, description: string,
              category: string, dateCreated: Date, dateModified: Date,
              isSubscribed: boolean, overallProgress: number,
              flashcards: FlashcardShort[]) {
    this.id = id;
    this.subLessonId = subLessonId;
    this.subCourseId = subCourseId;
    this.name = name;
    this.description = description;
    this.category = category;
    this.dateCreated = dateCreated;
    this.dateModified = dateModified;
    this.isSubscribed = isSubscribed;
    this.overallProgress = overallProgress;
    this.flashcards = flashcards;
  }
}
